code/modules/fissionreactor/misc.dm

```diff
@@ -94,3 +94,127 @@ boxes used for cargo orders to make my life easier.
 	fueldata.add_shit_to(URANIUM,units_of_storage,fueldata.fuel)
 	fueldata.rederive_stats()
 	fueldata.life=1
+
+
+/obj/item/weapon/fuelrod/randomized
+	icon_state="i_fuelrod"
+
+/obj/item/weapon/fuelrod/randomized/New()
+	..()	
+	var/list/mats_std	=	list(URANIUM,THORIUM) //conventional nuclear fuel
+	var/list/mats_uncommon=	list(PLUTONIUM,RADIUM) // exotic fuels or nuclear things
+	var/list/mats_rare	=	list(RADON) // reactor adjacent things or stuff notable to engineering
+	var/list/mats_exotic=	list(TRICORDRAZINE,DEGENERATECALCIUM) // for experimentation and learning
+	
+	var/current_units=fueldata.fuel.total_volume
+	while (current_units<units_of_storage)
+		var/r=rand()
+		var/material=null
+		var/amount_to_add=rand(1, ceil((units_of_storage-current_units)*0.8) )
+		if(amount_to_add>units_of_storage/6) //if we are adding a large amount, skew the randomness so that we don't give a huge amount of rare stuff. with current vars, max 15 units of rare things per roll.
+			r+=0.2
+		switch(r)
+			if(0.0 to 0.05) //  5%  skewed 0%
+				material=pick(mats_exotic)
+			if(0.05 to 0.2) // 15%  skewed 0%
+				material=pick(mats_rare)
+			if(0.2 to 0.5)  // 30%  skewed 30%
+				material=pick(mats_uncommon)
```

Comment:
Overlapping values in switch statements cause undefined behavior; please correct the intervals.